packages/medusa/src/loaders/__tests__/feature-flags.spec.ts

```diff
@@ -0,0 +1,133 @@
+import { resolve } from "path"
+import { mkdirSync, rmSync, writeFileSync } from "fs"
+
+import loadFeatureFlags from "../feature-flags"
+
+const distTestTargetDirectorPath = resolve(__dirname, "__ff-test__")
+
+const getFolderTestTargetDirectoryPath = (folderName: string): string => {
+  return resolve(distTestTargetDirectorPath, folderName)
+}
+
+const buildFeatureFlag = (
+  key: string,
+  defaultVal: string | boolean
+): string => {
+  const snakeCaseKey = key.replace(/-/g, "_")
+
+  return `
+    export default {
+      description: "${key} descr",
+      key: "${snakeCaseKey}",
+      env_key: "MEDUSA_FF_${snakeCaseKey.toUpperCase()}",
+      default_val: ${defaultVal},
+    }
+  `
+}
+
+describe("feature flags", () => {
+  const OLD_ENV = process.env
```

Comment:
suggestion: `const OLD_ENV = { ...process.env }` to avoid object reference sharing and unexpected alteration